src/main/java/org/logstash/dissect/Dissector.java

```diff
@@ -53,34 +56,58 @@ public Dissector(String mapping) {
         Collections.sort(saveableFields, new FieldComparator());
     }
 
-    public int dissect(byte[] source, Map<String, Object> map) {
-        int pos = dissectValues(source);
-        ValueResolver resolver = new ValueResolver(source, fieldValueRefMap);
+    public int dissect(byte[] source, Map<String, Object> keyValueMap) {
+        ResolveValue resolveValue = resolve(source);
```

Comment:
+1
To be honest, the only reason why I split it up was because IntelliJ complained about duplicate code in dissect(Map ..) and dissect(Event ..) methods.

Its easy to revert